plugins/modules/system/launchd.py

```diff
@@ -0,0 +1,482 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: launchd
+author:
+- Martin Migasiewicz (@martinm82)
+short_description:  Manage macOS services
+description:
+- Manage launchd services on target macOS hosts.
+options:
+    name:
+      description:
+      - Name of the service.
+      type: str
+      required: true
+    state:
+      description:
+      - C(started)/C(stopped) are idempotent actions that will not run
+        commands unless necessary.
+      - Launchd does not support C(restarted) nor C(reloaded) natively.
+        These will trigger a stop/start (restarted) or an unload/load
+        (reloaded).
+      - C(restarted) unloads and loads the service before start to ensure
+        that the latest job definition (plist) is used.
+      - C(reloaded) unloads and loads the service to ensure that the latest
+        job definition (plist) is used. Whether a service is started or
+        stopped depends on the content of the definition file.
+      type: str
+      choices: [ reloaded, restarted, started, stopped, unloaded ]
+    enabled:
+      description:
+      - Whether the service should start on boot.
+      - B(At least one of state and enabled are required.)
+      type: bool
+    force_stop:
+      description:
+      - Whether the service should not be restarted automatically by launchd.
+      - Services might have the 'KeepAlive' attribute set to true in a launchd configuration.
+        In case this is set to true, stopping a service will cause that launchd starts the service again.
+      - Set this option to C(yes) to let this module change the 'KeepAlive' attribute to false.
+      type: bool
+      default: no
+notes:
+- One option other than name is required.
+requirements:
+- A system managed by launchd
+- The plistlib python library
+'''
+
+EXAMPLES = r'''
+- name: Make sure spotify webhelper is started
+  community.general.launchd:
+    name: com.spotify.webhelper
+    state: started
+  become: yes
+
+- name: Deploy custom memcached job definition
+  template:
+    src: org.memcached.plist.j2
+    dest: /Library/LaunchDaemons/org.memcached.plist
+  become: yes
+
+- name: Run memcached
+  community.general.launchd:
+    name: org.memcached
+    state: started
+  become: yes
+
+- name: Stop memcached
+  community.general.launchd:
+    name: org.memcached
+    state: stopped
+  become: yes
+
+- name: Stop memcached
+  community.general.launchd:
+    name: org.memcached
+    state: stopped
+    force_stop: yes
+  become: yes
+
+- name: Restart memcached
+  community.general.launchd:
+    name: org.memcached
+    state: restarted
+  become: yes
+
+- name: Unload memcached
+  community.general.launchd:
+    name: org.memcached
+    state: unloaded
+  become: yes
+'''
+
+RETURN = r'''
+status:
+    description: metadata about service status
+    returned: always
+    type: dict
+    sample:
+        {
+            "current_pid": "-",
```

Comment:
```suggestion
            "current_pid": "",
```
IMO better, just a recommendation

Reply:
pids are always integers AFAIK. maybe to return them instead of strings and something like None or Null if possible (i don't know it's technically possible or not)